rad/src/error.rs

```diff
@@ -1,87 +1,100 @@
 //! Error type definitions for the RAD module.
 
-use reqwest;
-
-use failure::Fail;
-use std::fmt;
-pub use witnet_util::error::{WitnetError, WitnetResult};
-
-/// RAD Error
-#[derive(Debug, PartialEq, Fail)]
-#[fail(display = "{} : {}", kind, msg)]
-pub struct RadError {
-    /// Error kind.
-    kind: RadErrorKind,
-    /// Error message (likely passed from the originating exception).
-    msg: String,
-}
-
-impl RadError {
-    /// Create a RAD error based on error kind, context and message.
-    pub fn new(kind: RadErrorKind, msg: String) -> Self {
-        Self { kind, msg }
-    }
-
-    /// Query the specific RadErrorKind case for a RadError
-    pub fn kind(&self) -> &RadErrorKind {
-        &self.kind
-    }
-}
-
-impl From<reqwest::Error> for RadError {
-    fn from(x: reqwest::Error) -> RadError {
-        RadError::new(RadErrorKind::Http, x.to_string())
-    }
-}
+use failure::{self, Fail};
+use rmpv::{Integer, Value};
 
 /// RAD errors.
-#[derive(Debug, PartialEq)]
-pub enum RadErrorKind {
-    /// Failed to encode or decode a RADON type into / from bytes
-    EncodeDecode,
+#[derive(Debug, PartialEq, Fail)]
+pub enum RadError {
+    /// Failed to decode a type from other
+    #[fail(display = "Failed to decode {} from {}", from, to)]
```

Comment:
```suggestion
    #[fail(display = "Failed to decode {} from {}", to, from)]
```

ParseInt: Decode from string to int